include/deal.II/numerics/rtree.h

```diff
@@ -498,9 +498,26 @@ RTree<typename ContainerType::size_type,
       IndexableGetterFromIndices<ContainerType>>
 pack_rtree_of_indices(const ContainerType &container)
 {
+  // We need an array that holds the indices we want to pack. The rtree
+  // implementation in BOOST, for reasons not entirely clear, insists
+  // on using a reference to the elements of the range. This is fine if
+  // the indices are stored in a container, but not if the range of
+  // indices is given by a std::ranges::iota_view which does not offer
+  // a nested 'reference' type, and this then trips up BOOST rtree.
+  //
+  // On the other hand, BOOST's own iota_view implementation offers the
+  // 'reference' nested type, so use it if we don't build on the C++20
+  // iota_view class.
+#  ifdef DEAL_II_HAVE_CXX20
+  std::vector<typename ContainerType::size_type> indices(container.size());
+  for (typename ContainerType::size_type i = 0; i < container.size(); ++i)
+    indices[i] = i;
```

Comment:
I'd prefer if we just keep this branch unconditionally. Otherwise, this looks pretty confusing since we are using `std_cxx20` in the non-C++20 branch which then always uses boost.

Reply:
Sure, fair point.